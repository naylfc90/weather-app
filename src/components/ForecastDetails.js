import React from "react";
import PropTypes from "prop-types";
import moment from "moment";

import "../styles/ForecastDetails.css";

const ForecastDetails = ({ forecast }) => {
  const { date, temperature, humidity, wind } = forecast;
  return (
    <div className="forecast-details">
      <div className="forecast-details__date" data-testid="forecast-date">
        <h2>{moment(date).format("ddd Do MMM")}</h2>
      </div>

      <div
        className="forecast-details__temperature"
        data-testid="forecast-temperature"
      >
        Max Temperature: {temperature.max}
        &deg;C
        <br />
        Min Temperature: {temperature.min}
        &deg;C
      </div>

      <div className="forecast-details__humidity">Humidity: {humidity}%</div>

      <div className="forecast-details__wind">
        Wind: {wind.speed} mph
        <br />
        Wind Direction: {wind.direction.toUpperCase()}
      </div>
    </div>
  );
};

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number.isRequired,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }).isRequired,
    humidity: PropTypes.number.isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};

export default ForecastDetails;
