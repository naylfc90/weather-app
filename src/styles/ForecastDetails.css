.forecast-details {
  text-align: center;
  margin-top: 25px;
  margin-left: 33%;
  margin-right: 33%;
  background: #86bbd8;
  padding: 15px;
  border: solid 2px;
  border-radius: 10px;
}
