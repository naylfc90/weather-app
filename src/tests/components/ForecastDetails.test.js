import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    temperature: {
      min: 12,
      max: 22,
    },
    humidity: 30,
    wind: {
      speed: 20,
      direction: "n",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("render correct values for props", () => {
    const { getByTestId } = render(<ForecastDetails forecast={validProps} />);

    expect(getByTestId("forecast-date")).toHaveClass("forecast-details__date");
    expect(getByTestId("forecast-temperature")).toHaveClass(
      "forecast-details__temperature"
    );
  });
});
