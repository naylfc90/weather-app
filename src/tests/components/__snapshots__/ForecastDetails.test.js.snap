// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
      data-testid="forecast-date"
    >
      <h2>
        Thu 1st Jan
      </h2>
    </div>
    <div
      class="forecast-details__temperature"
      data-testid="forecast-temperature"
    >
      Max Temperature: 22°C
      <br />
      Min Temperature: 12°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 30%
    </div>
    <div
      class="forecast-details__wind"
    >
      Wind: 20 mph
      <br />
      Wind Direction: N
    </div>
  </div>
</DocumentFragment>
`;
